src/pkg/repository/dao/dao.go

```diff
@@ -166,7 +166,7 @@ func (d *dao) NonEmptyRepos(ctx context.Context) ([]*models.RepoRecord, error) {
 		return nil, err
 	}
 
-	sql := fmt.Sprintf(`select distinct r.* from repository as r LEFT JOIN tag as t on r.repository_id = t.repository_id where t.repository_id is not null;`)
+	sql := `select * from repository where repository_id in (select distinct repository_id from tag)`
```

Comment:
no need distinct

Reply:
'distinct' can reduce the duplicate key from tag table. it'a a cherry-pick, how about let's keep it?